### PR TITLE
Remove all shell presubmits

### PR DESCRIPTION
**Pull request by miekg**
This ports the shell presubmit to Go and makes them better; esp the
Errorf/Logf/Fatalf one because it actually parses the code and works of
the AST.

These error will now show up in the travis/circle CI runs where they
belong.

### CODE REVIEW

**Review by yongtang — APPROVED** [review]
Nice! 👍 /lgtm

File: test/presubmit_test.go [hunk 1]
```diff
@@ -75,3 +78,133 @@ func hasHyphen(path string, info os.FileInfo, _ error) error {
...
+	}
+	se, ok := ce.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return l
+	}
+	id, ok := se.X.(*ast.Ident)
+	if !ok {
+		return l
+	}
+	if id.Name != "t" { //t *testing.T
```
Comment (stp-ip) [inline test/presubmit_test.go:135]: Is that comment intentional? Just trying to understand the specifics here.

**Review by stp-ip — APPROVED** [review]
Besides the question.
/lgtm

File: test/presubmit_test.go (hunk 1 above)
Comment (miekg) [reply test/presubmit_test.go:135]: yes, "t" signals that Errorf is called from a var named `t` which we assume is a *testing.T (although the type isn't checked)

Comment (stp-ip) [reply test/presubmit_test.go:135]: Thanks for the context. Not sure I like that unchecked assumption, but ok with just the comment.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] Remove all sh..." ]
>Thanks for the context. Not sure I like that unchecked assumption, but ok with just the comment.

"it's better than it was" - as it was only a grep

**Pull request merged**